crates/terminal_view/src/terminal_panel.rs

```diff
@@ -315,10 +315,14 @@ impl TerminalPanel {
             return;
         };
 
-        let mut command = std::mem::take(&mut spawn_task.command);
+        let old_command = std::mem::take(&mut spawn_task.command);
```

Comment:
Is this change likely to break user tasks in the wild?

Reply:
Not sure, how? 

In the previous logic, simplified, we've combined `command` and `args` into one large string and did `$SHELL -c $COMBINED_STRING` on it.

Now, we do the same, but escape all shell-related things before doing that. I imagined that would not break existing scripts, but fix some of those that did not work.
For me, it was the `python -c` one:
```
args: [
    "-i",
    "-c",
    "python3 -c \"exec(r'''def foo():\n    print(\"Hello, fun!\")\n\nif __name__ == \"__main__\":\n    print(\"Hello, World!\")\n\n''')\"",
],
```
produces
```
File "<string>", line 2
      print(Hello,
                  ^
SyntaxError: EOF while scanning triple-quoted string literal
```

Reply:
Yeah idk, I don't see how it would break anything either, but I've figured that maybe it's worth asking. :p